zerver/lib/bot_lib.py

```diff
@@ -83,7 +91,26 @@ def identity(self) -> BotIdentity:
         return BotIdentity(self.full_name, self.email)
 
     def react(self, message: Dict[str, Any], emoji_name: str) -> Dict[str, Any]:
-        return {}  # Not implemented
+        # Currently, instead of raising exceptions, we return dictionaries containing the error
+        # message for bot handlers. But in the future, these should be replaced by python exceptions.
+        if "id" not in message:
+            generate_error_msg(_('Expected to find the "id" required key.'))
+
+        try:
+            emoji_code, reaction_type = emoji_name_to_emoji_code(
+                self.user_profile.realm, emoji_name
+            )
+
+            do_add_reaction(
+                self.user_profile,
+                Message.objects.filter(id=message["id"])[0],
+                emoji_name,
+                emoji_code,
+                reaction_type,
+            )
+        except JsonableError as e:
+            return generate_error_msg(e.msg)
```

Comment:
I don't think `EmbeddedBotHandler` should handle exceptions after the latest discussions.

Reply:
OK, then I guess we will return an empty dictionary instead.

Reply:
It is the `worker` that should do that if required.